src/release.js

```diff
@@ -507,6 +507,33 @@ module.exports = function taoExtensionReleaseFactory(params = {}) {
             }
         },
 
+        /**
+         * Verify that the version that we are going to release is valid
+         * - is the same on branch name and manifest
+         * - is bigger than current release branch version.
+         */
+        async verifyReleasingBranch() {
+            log.doing('Validating releasing branch.');
+
+            // Cross check releasing branch version with manifest version
+            await gitClient.checkout(data.releasingBranch);
+            const releasingBranchManifest = await taoInstance.parseManifest(`${data.extension.path}/manifest.php`);
+            if (compareVersions(releasingBranchManifest.version, data.version) == 0 ) {
+                log.doing(`Branch ${data.releasingBranch} has valid manifest.`);
+            } else {
+                log.exit(`Mismatch versions found between branch '${data.releasingBranch}' and manifest version '${releasingBranchManifest.version}'.`);
```

Comment:
I would like the exit message to be more precise.
"Branch foo-1.2.3 cannot be released because its branch name does not match its own manifest version (1.2.2)"